### DISCUSSION TITLE
Should we add awesome-webgpu to sindresorhus/awesome?

### DISCUSSION THREAD

**Discussion by ferminLR** [mikbry/awesome-webgpu#44] [General]
What the title says. I was expecting it to be already included, but somehow it isn't yet. I think the list is already populated enough, and WebGPU should gain traction now that is more widely supported. The list is worthy of being feature in sindresorhus/awesome and maybe reach more people thanks to that.

I offer myself to make changes, if necessary, for awesome-webgpu to be accepted. What do you think?

**Comment by mikbry** [comment]
I love the idea. Yes you could do it. Thanks!

**Comment by ferminLR** [comment]
I've already made a PR with several small changes. There's some work left to do:

- The default branch should be renamed `main`
- Add `awesome` to Github topics.
- Run [awesome-lint](https://github.com/sindresorhus/awesome-lint) on the list
- Go through the items in the list checking everything is still online, non-deprecated.
- Add descriptions to each item in the list, unless the title is descriptive enough.

I can't make the first two; @mikbry I think only you can do them.
Going through the entire list to complete the last two point will take me some time, I could use some help there :)

**Comment by mikbry** [comment]
I made first two changes.
Thanks for the PR @ferminLR

**Comment by mikbry** [comment]
I just push some fixes for awesome-lint, it now pass without any errors

**Comment by mikbry** [comment]
Hi @ferminLR are you still planning to submit awseome-webgpu on sindresorhus/awesome ? Need any help ?

**Reply by ferminLR** [reply]
Yes! I am still planning to do it, it's just I've been very busy lately and couldn't dedicate a lot of time to this.
I am currently going through the list adding descriptions and removing what is deprecated.

**Reply by mikbry** [reply]
Great !

**Comment by ferminLR** [comment]
I just made a PR with some changes, there's still some work to do though.

I have a question @mikbry, there are some links to old projects that don't work anymore as the WebGPU API has changed over the years. Should we put them in a new "Old WebGPU API" section (no idea how I would call the section) or we just remove them?

**Comment by mikbry** [comment]
Thanks @ferminLR  for your PR, some minor errors to fix.

Remove them is better, it should stay a quality list not an exhaustive list.

**Reply by ferminLR** [reply]
I agree

**Comment by ferminLR** [comment]
I just made a new PR with my final changes. For me, the list is ready to add to `sindresorhus/awesome`.

**Reply by mikbry** [reply]
Great job @ferminLR ! so let's go add it to sindresorhus/awesome ;-)

**Reply by ferminLR** [reply]
PR created!

**Reply by ferminLR** [reply]
PR not accepted! Not sure why, though, I think we met all the requirements ¯\\\_(ツ)_/¯

**Reply by mikbry** [reply]
Same I don't understand...

**Reply by mikbry** [reply]
Here is the complete check list, could you validate again, thanks for your work

- [ ] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- [ ] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [ ] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions. Comments pointing out lint violation are allowed, but does **not** count as a review.
- [ ] You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- [ ] This pull request has a title in the format `Add Name of List`. It should not contain the word `Awesome`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- [ ] Your entry here should include a short description of the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb. It should not contain the name of the list.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- [ ] Your entry should be added at the bottom of the appropriate category.
- [ ] The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- [ ] No blockchain-related lists.
- [ ] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [ ] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- [ ] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- [ ] The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- [ ] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- [ ] It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- [ ] The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- [ ] The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- [ ] Non-generated Markdown file in a GitHub repo.
- [ ] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [ ] Not a duplicate. Please search for existing submissions.
- [ ] Only has awesome items. Awesome lists are curations of the best, not everything.
- [ ] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- [ ] Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to a maximum of half the width of the original image.
	- Don't include both a title saying `Awesome X` and a logo with `Awesome X`. You can put the header image in a `#` (Markdown header) or `<h1>`.
- [ ] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [ ] Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- [ ] Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- [ ] Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- [ ] Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. The casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- [ ] All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- [ ] Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [ ] Does not use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- [ ] Does not include a CI (e.g. GitHub Actions) badge.<br>You can still use a CI for linting, but the badge has no value in the readme.
- [ ] Does not include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

**Reply by mikbry** [reply]
@ferminLR did you check this two points :
- To verify that you've read all the guidelines, please comment on your pull request with just the word unicorn.
- You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).

I work on rewriting all entries description

**Reply by ferminLR** [reply]
I did the review of other PRs, but I completely missed doing the comment with the word unicorn! thanks for the tip.

I just did the comment, let's see if it's approved now.
